### DISCUSSION TITLE
how to catch parameter errors from endpoint functions??

### DISCUSSION THREAD

**Discussion by ameerfayiz** [serverpod/serverpod#1064] [Q&A] [answered]
for a given sample function :

```
Future<ServerResponse> sample(Session session, Book book) async {
 ....
 ....  
}
```

if i called the endpoint from postman , without giving the book object, or with a wrong json which cannot be deserialized into Book object, i get "internal error" from the server. i want to catch that error and give out proper server response back to the client. how to do it?? do i have to manually de-serialize the incoming object inside a try catch instead of making it non-nullable in the endpoint function parameter ??

**Comment by starklord** [comment]
Hi, have you looked at this https://docs.serverpod.dev/concepts/exceptions
Maybe if you create your own Exception object and catch on the client as you wish

**Reply by ameerfayiz** [reply]
way to catch it from server endpoint?? because internal error can occur due to many other reasons right. so client cannot parse the type of error unless i pass the error is due to json parsing of the endpoint parameters

**Comment by vlidholt** [comment, marked as answer]
If you pass the wrong set of parameters to Serverpod, it will obviously give you a result that is an error as it validates the parameters for you. The main idea is that you use the client package generated by Serverpod and you will never run into this issue (if you do for some reason, you will get an exception that you handle like any other exception).

If you are writing your own client code, that seems like a lot of extra work and the only reason is if you do it outside of Flutter which is sort of outside the scope for Serverpod (at least for now).
